end_to_end/end_to_end_suite_test.go

```diff
@@ -2657,4 +2657,175 @@ LANGUAGE plpgsql NO SQL;`)
+					INSERT INTO part_c SELECT a, 0, a, 'c' FROM generate_series(1, 30)a;
+					INSERT INTO test2_part SELECT a, 35, a, 'test' FROM generate_series(1, 40)a;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir)
+			metadataFileContents := string(getMetdataFileContents(backupDir, timestamp, "metadata.sql"))
+			Expect(metadataFileContents).To(ContainSubstring("test_part"))
+			Expect(metadataFileContents).To(ContainSubstring("part_a"))
+			Expect(metadataFileContents).To(ContainSubstring("part_b"))
+			Expect(metadataFileContents).To(ContainSubstring("part_c"))
+			Expect(metadataFileContents).To(ContainSubstring("test2_part"))
+
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb")
+
+			assertDataRestored(restoreConn, map[string]int{
+				"public.test_part":  30,
+				"public.part_a":     10,
+				"public.part_b":     20,
+				"public.part_c":     30,
+				"public.test2_part": 40,
+			})
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("backup the partition whose parent is in the extension with '--leaf-partition-data' flag", func() {
+			testhelper.AssertQueryRuns(backupConn, `
+					CREATE TABLE test_part PARTITION OF t_part FOR VALUES FROM (10) TO (20) PARTITION BY LIST(c);
+					CREATE TABLE part_a PARTITION OF test_part FOR VALUES IN ('a');
+					CREATE TABLE part_b PARTITION OF test_part FOR VALUES IN ('b');
+
+					CREATE TABLE part_c PARTITION OF t_part_1_prt_1 FOR VALUES IN ('c');
+
+					CREATE TABLE test2_part PARTITION OF t_part FOR VALUES FROM (30) TO (40);
+
+					INSERT INTO part_a SELECT a, 15, a, 'a' FROM generate_series(1, 10)a;
+					INSERT INTO part_b SELECT a, 16, a, 'b' FROM generate_series(1, 20)a;
+					INSERT INTO part_c SELECT a, 0, a, 'c' FROM generate_series(1, 30)a;
+					INSERT INTO test2_part SELECT a, 35, a, 'test' FROM generate_series(1, 40)a;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir, "--leaf-partition-data")
+			metadataFileContents := string(getMetdataFileContents(backupDir, timestamp, "metadata.sql"))
+			Expect(metadataFileContents).To(ContainSubstring("test_part"))
+			Expect(metadataFileContents).To(ContainSubstring("part_a"))
+			Expect(metadataFileContents).To(ContainSubstring("part_b"))
+			Expect(metadataFileContents).To(ContainSubstring("part_c"))
+			Expect(metadataFileContents).To(ContainSubstring("test2_part"))
+
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb")
+
+			assertDataRestored(restoreConn, map[string]int{
+				"public.test_part":  30,
+				"public.part_a":     10,
+				"public.part_b":     20,
+				"public.part_c":     30,
+				"public.test2_part": 40,
+			})
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("ignore partitions specified in exclude-table during backup", func() {
```

Comment:
This tests and tests below miss the part with restoration. It's important to check that script gpbackup was generated can be successfully restored with desired set of data.

Reply:
fixed